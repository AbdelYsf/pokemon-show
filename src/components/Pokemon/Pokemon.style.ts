import {Card, CardActions, CardMedia, styled, Typography,} from "@mui/material";

export const StyledCardActions = styled(CardActions)`
  display: flex;
  align-items: center;
  flex-direction: column;
`;

export const StyledCard = styled(Card)`
  background-color: #faebec;
  margin: 20px;
  width: 260px;
  border-radius: 25px;

  :hover {
    transform: scale(1.03);
  }

  transition: all 0.2s;
`;

export const StyledCardMedia = styled(CardMedia)`
  object-fit: contain;
  max-height: 100%;
  max-width: 100%;
` as typeof CardMedia;

export const StyledTypography = styled(Typography)`
  display: flex;
  justify-content: center;
  text-transform: capitalize;
` as typeof Typography;
