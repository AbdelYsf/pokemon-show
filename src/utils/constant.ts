export const POKEMON_BALL_ICON_URL =
    "https://upload.wikimedia.org/wikipedia/commons/thumb/5/53/Pok%C3%A9_Ball_icon.svg/2052px-Pok%C3%A9_Ball_icon.svg.png";

export const POKEMON_API_ROOT_URL = "https://pokeapi.co/api/v2"
export const URL_ID = `{ID}`;

export const OFFICIAL_ARTWORK = `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/${URL_ID}.png`

export const MAX_STATS_VALUE = 255;