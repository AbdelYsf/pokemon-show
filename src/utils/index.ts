export const getPokemonIdFromUrl = (url: string) => {
    const result = url.split("/")
    return result[result.length - 2]
}

export const convertDmToCm = (value?: number) => {
    // convert decimeter to centimeter
    return value ? value * 10 : 0
}

export const convertHgToKg = (value?: number) => {
    // convert hectogramme to kilogramme
    return value ? value / 10 : 0
}