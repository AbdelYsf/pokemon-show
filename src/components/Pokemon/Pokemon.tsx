import { Avatar, Button } from "@mui/material";
import { PokemonProps } from "../../types";
import {
  StyledCard,
  StyledCardActions,
  StyledCardMedia,
  StyledTypography,
} from "./Pokemon.style";
import { POKEMON_BALL_ICON_URL } from "../../utils/constant";

const Pokemon = (props: PokemonProps) => {
  return (
    <StyledCard>
      <StyledCardMedia
        component="img"
        height="260"
        image={props.imageUrl}
        alt={props.name}
      />
      <StyledCardActions>
        <StyledTypography
          gutterBottom
          variant="h4"
          component="div"
          fontFamily="Rampart One"
        >
          {props.name}
        </StyledTypography>
      </StyledCardActions>
      <StyledCardActions>
        <Button
          size="large"
          startIcon={<Avatar src={POKEMON_BALL_ICON_URL}></Avatar>}
        ></Button>
      </StyledCardActions>
    </StyledCard>
  );
};

export default Pokemon;
