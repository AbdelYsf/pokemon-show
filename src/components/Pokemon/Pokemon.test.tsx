import React from "react";
import {render} from "@testing-library/react";
import Pokemon from "./Pokemon";
import {screen} from "@testing-library/react";
import {POKEMON_BALL_ICON_URL} from "../../utils/constant";
import userEvent from "@testing-library/user-event";

describe("Pokemon", () => {
    // fixtures
    const pokemonName = "Pikachu";
    const pokemonImageUrl = "url";

    test("should render Pokemon", () => {
        render(<Pokemon name={pokemonName} imageUrl={pokemonImageUrl}/>);
        expect(screen.getByText(pokemonName)).toBeInTheDocument();
        expect(screen.getByRole("button")).toBeInTheDocument();
        const images = screen.getAllByRole("img");
        expect(images).toHaveLength(2);
        expect(images[0]).toHaveAttribute("src", "url");
        expect(images[1]).toHaveAttribute("src", POKEMON_BALL_ICON_URL);
    });

    test("renders the modal of pokemon details", async () => {
        const user = userEvent.setup()
        render(<Pokemon name={pokemonName} imageUrl={pokemonImageUrl}/>);
        const button = screen.getByRole("button");
        await user.click(button);
        // TODO: add assertions
    });

    test("should match snapshot", async () => {
        const container = render(<Pokemon name={pokemonName} imageUrl={pokemonImageUrl}/>);
        expect(container).toMatchSnapshot()
    });
});
