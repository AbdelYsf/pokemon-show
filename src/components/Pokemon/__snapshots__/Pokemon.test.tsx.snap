// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1tujc56-MuiPaper-root-MuiCard-root"
      >
        <img
          alt="Pikachu"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-t7s8y2-MuiCardMedia-root"
          height="260"
          src="url"
        />
        <div
          class="MuiCardActions-root MuiCardActions-spacing css-72hqeq-MuiCardActions-root"
        >
          <div
            class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1m4q058-MuiTypography-root"
          >
            Pikachu
          </div>
        </div>
        <div
          class="MuiCardActions-root MuiCardActions-spacing css-72hqeq-MuiCardActions-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeLarge MuiButton-textSizeLarge css-kpm0tx-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeLarge css-htszrh-MuiButton-startIcon"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
              >
                <img
                  class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                  src="https://upload.wikimedia.org/wikipedia/commons/thumb/5/53/Pok%C3%A9_Ball_icon.svg/2052px-Pok%C3%A9_Ball_icon.svg.png"
                />
              </div>
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1tujc56-MuiPaper-root-MuiCard-root"
    >
      <img
        alt="Pikachu"
        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-t7s8y2-MuiCardMedia-root"
        height="260"
        src="url"
      />
      <div
        class="MuiCardActions-root MuiCardActions-spacing css-72hqeq-MuiCardActions-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1m4q058-MuiTypography-root"
        >
          Pikachu
        </div>
      </div>
      <div
        class="MuiCardActions-root MuiCardActions-spacing css-72hqeq-MuiCardActions-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeLarge MuiButton-textSizeLarge css-kpm0tx-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeLarge css-htszrh-MuiButton-startIcon"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
            >
              <img
                class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                src="https://upload.wikimedia.org/wikipedia/commons/thumb/5/53/Pok%C3%A9_Ball_icon.svg/2052px-Pok%C3%A9_Ball_icon.svg.png"
              />
            </div>
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
